app/backend/core/modelhelper.py

```diff
@@ -43,7 +43,9 @@ def num_tokens_from_messages(message: dict[str, str], model: str) -> int:
 
 
 def get_oai_chatmodel_tiktok(aoaimodel: str) -> str:
+    message = "Expected Azure OpenAI ChatGPT model name"
     if aoaimodel == "" or aoaimodel is None:
-        raise ValueError("Expected AOAI chatGPT model name")
-
-    return AOAI_2_OAI.get(aoaimodel)
+        raise ValueError(message)
+    if aoaimodel not in AOAI_2_OAI and aoaimodel not in MODELS_2_TOKEN_LIMITS:
```

Comment:
This is the potential issue I mentioned in the PR description. I think we want this function to also return the model if its not a name that has to be mapped, so that num_tokens_from_messages doesn't error for models like gpt-4, right?

Reply:
Thanks for adding this.